core/src/main/java/com/amplifyframework/auth/AuthException.java

```diff
@@ -396,6 +396,41 @@ public LimitExceededException(Throwable cause) {
         }
     }
 
+    /**
+     * Could not find multi-factor authentication (MFA) method in AWS Cognito.
+     */
+    public static class MFAMethodNotFoundException extends AuthException {
+        private static final long serialVersionUID = 1L;
+        private static final String MESSAGE = "Could not find multi-factor authentication (MFA) method.";
+        private static final String RECOVERY_SUGGESTION = "Configure multi-factor authentication using Amplify CLI or AWS Cognito console.";
+
+        /**
+         * Default message/recovery suggestion with a cause.
+         * @param cause The original error.
+         */
+        public MFAMethodNotFoundException(Throwable cause) {
+            super(MESSAGE, cause, RECOVERY_SUGGESTION);
+        }
+    }
+
+    /**
+     *
+     */
+    public static class NotAuthorizedException extends AuthException {
+        private static final long serialVersionUID = 1L;
+        private static final String MESSAGE = "";
```

Comment:
The message should not be empty.
```suggestion
        private static final String MESSAGE = "Failed since user is not authorized.";
```